src/QsCompiler/LanguageServer/EditorState.cs

```diff
@@ -29,7 +28,7 @@ internal class EditorState : IDisposable
         public void Dispose() => this.projects.Dispose();
 
         private readonly Action<PublishDiagnosticParams> publish;
-        private readonly Action<string, Dictionary<string, string?>, Dictionary<string, int>> sendTelemetry;
+        private readonly Action<string, Dictionary<string, string?>, Dictionary<string, int>>? sendTelemetry;
```

Comment:
As before, I suggest using a well defined delegate here to make it clear what each input parameter of the action mean.
And maybe make this an event instead of a standard field.